include/nbl/video/utilities/IUtilities.h

```diff
@@ -729,7 +757,56 @@ class NBL_API IUtilities : public core::IReferenceCounted
         core::smart_refctd_ptr<CScanner> m_scanner;
     };
 
-}
+class NBL_API ImageRegionIterator
+{
+public:
+    ImageRegionIterator(
+        const core::SRange<const asset::IImage::SBufferCopy>& copyRegions,
+        IPhysicalDevice::SQueueFamilyProperties queueFamilyProps,
```

Comment:
what is it exactly do you need the queue family properties for?

Reply:
minImageTransferGranularity, I guess I could have taken that directly. I made a choice to take the whole struct to emphasize that the regionIterator depends on the queue family you're going to submit the image copies to.

Reply:
Oh I use it for other stuff, deciding on some alignments based on queue family caps.